test/validate/validate.go

```diff
@@ -154,77 +167,121 @@ func (v *Validator) validateIPs(ctx context.Context, stateFileIps stateFileIpsFu
+	log.Print("Validating Node properties")
 
 	for index := range nodes.Items {
 		nodeName := nodes.Items[index].ObjectMeta.Name
 		// check nodes status;
 		// nodes status should be ready after cluster is created
 		nodeConditions := nodes.Items[index].Status.Conditions
 		if nodeConditions[len(nodeConditions)-1].Type != corev1.NodeReady {
-			return errors.Wrapf(err, "node %s status is not ready", nodeName)
+			return errors.Errorf("node %s status is not ready", nodeName)
 		}
 
 		// get node labels
 		nodeLabels := nodes.Items[index].ObjectMeta.GetLabels()
 		for key := range nodeLabels {
-			if label, ok := dualstackOverlayNodeLabels[key]; ok {
+			if label, ok := labels[key]; ok {
 				log.Printf("label %s is correctly shown on the node %+v", key, nodeName)
 				if label != overlayClusterLabelName {
-					return errors.Wrapf(err, "node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
+					return errors.Errorf("node %s overlay label name is wrong; expected label:%s but actual label:%s", nodeName, overlayClusterLabelName, label)
 				}
 			}
 		}
 
-		// check if node has two internal IPs(one is IPv4 and another is IPv6)
+		// check if node has corret number of internal IPs
 		internalIPCount := 0
 		for _, address := range nodes.Items[index].Status.Addresses {
 			if address.Type == "InternalIP" {
 				internalIPCount++
 			}
 		}
-		if internalIPCount != 2 { //nolint
-			return errors.Wrap(err, "node does not have two internal IPs")
+		if internalIPCount != expectedIPCount {
+			return errors.Errorf("node does not expected number of IPs %d", expectedIPCount)
+		}
+	}
+	return nil
+}
+
+func (v *Validator) ValidateV4OverlayControlPlane(ctx context.Context) error {
+	nodes, err := acnk8s.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap[v.os])
+	if err != nil {
+		return errors.Wrap(err, "failed to get node list")
+	}
+
+	if err := validateNodeProperties(nodes, v4OverlayNodeLabels, IPv4ExpectedIPCount); err != nil {
+		return errors.Wrap(err, "failed to validate IPv4 overlay node properties")
+	}
+
+	if v.os == "windows" {
+		if err := validateHNSNetworkState(ctx, nodes, v.clientset, v.config); err != nil {
+			return errors.Wrap(err, "failed to validate IPv4 overlay HNS network state")
 		}
 	}
 
 	return nil
 }
 
 func (v *Validator) ValidateDualStackControlPlane(ctx context.Context) error {
-	if err := v.validateDualStackNodeProperties(ctx); err != nil {
+	nodes, err := acnk8s.GetNodeListByLabelSelector(ctx, v.clientset, nodeSelectorMap[v.os])
+	if err != nil {
+		return errors.Wrap(err, "failed to get node list")
+	}
+
+	if err := validateNodeProperties(nodes, dualstackOverlayNodeLabels, DualstackExpectedIPCount); err != nil {
 		return errors.Wrap(err, "failed to validate dualstack overlay node properties")
 	}
 
 	return nil
 }
 
+func (v *Validator) RestartKubeProxyService(ctx context.Context) error {
+	nodes, err := acnk8s.GetNodeList(ctx, v.clientset)
```

Comment:
We can filter based on windows here and then remove line 266 ?

Reply:
Do you mean, inside of `GetNodeList()` ? Or just outside on the results ? 
If outside on the results, I would argue that's just the same as skipping non-windows nodes in the for loop